src/index.ts

```diff
@@ -5,7 +5,39 @@
  * @author Kent C. Dodds <me@kentcdodds.com> (https://kentcdodds.com)
  */
 import removeAccents from 'remove-accents'
+import type {Merge} from 'type-fest'
 
+type KeyAttributes = {
+  threshold?: number
+  maxRanking: number
+  minRanking: number
+}
+type RankingInfo = {
+  rankedValue: string
+  rank: number
+  keyIndex: number
+  keyThreshold: number | undefined
+}
+
+declare function valueGetterKey<ItemType>(item: ItemType): string
+declare function baseSortFn<ItemType>(
+  a: Merge<RankingInfo, {item: ItemType; index: number}>,
```

Comment:
with interfaces (or even just `&`, but the interface API is cleaner IMO), you can get rid of this dependency:

```ts
interface RankingInfo {
  rankedValue: string
  rank: number
  keyIndex: number
  keyThreshold: number | undefined
}
interface IndexedItem<ItemType> {
  item: ItemType
  index: number
}
interface RankedItem<ItemType> extends RankingInfo, IndexedItem<ItemType> {}

declare function baseSortFn<ItemType>(
  a: RankedItem<ItemType>,
  b: RankedItem<ItemType>,
): number
```